packages/react-dom/src/__tests__/ReactCompositeComponent-test.js

```diff
@@ -5,6 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  *
  * @emails react-core
+ * @flow
```

Comment:
Why type tests?

Reply:
I don't know. You tell me :-)

In all seriousness, it seems like because of the [`shallowEqual`](https://github.com/facebook/react/blob/1856f59ade6524b72f9ea720eab045572503bac9/packages/react-dom/src/__tests__/ReactCompositeComponent-test.js#L30) function which seems like was just copied into the test suite for some reason and (in)conveniently just included types. Could probably just use the one at `packages/shared/shallowEqual.js` instead?